### ISSUE TITLE
`void()` incorrectly trapped in `nf_checkDSLcode`

### ISSUE THREAD

**Issue by perrydv** [nimble-dev/nimble#381]
```{r}
foo <- nimbleFunction(
        run = function(x = double(1)) {
            return()
            returnType(void())
        })
```
This should not warn the user that `void` will need to be defined.

Side note: Do we want to use `call. = FALSE` when issuing the warning [here](https://github.com/nimble-dev/nimble/blob/88368a57f0bf7a1cc0a681d7fed0fe74bc538d49/packages/nimble/R/RCfunction_core.R#L140)?

**Comment by perrydv** [comment]
@paciorek just making sure you noticed this as a small fix for 0.6-5

**Comment by paciorek** [comment]
done on both fronts.
